snapcraft/_store.py

```diff
@@ -184,6 +183,30 @@ def register(snap_name, is_private=False):
         snap_name))
 
 
+def sign_build(snap_filename, key_name='default', local=False):
+    if not repo.is_package_installed('snapd'):
+        raise EnvironmentError(
+            'The snapd package is not installed. In order to use '
+            '`sign-build`, you must run `apt install snapd`.')
+
+    if not os.path.exists(snap_filename):
+        raise FileNotFoundError(
+            'The file {!r} does not exist.'.format(snap_filename))
+
+    # we pass account_info to sign_build so we call it only once
+    # for both generating/signing and pushing the assertion data
+    store = storeapi.StoreClient()
+    account_info = store.get_account_information()
+
+    snap_yaml = _get_data_from_snap_file(snap_filename)
+    snap_name = snap_yaml['name']
+    grade = snap_yaml['grade']
+
+    with _requires_login():
+        store.sign_build(
+            account_info, snap_name, snap_filename, grade, key_name, local)
```

Comment:
Right, account-info requires login, which means it can't ever create and sign a snap-build w/o talking to the Store. This is less than ideal and will have to fixed eventually (caching it locally) and therefore, should not 'leak' to `generate_snap_build()` or `push_snap_build()`. Let's try to solve this here and pass specific info down.